soroban-sdk/src/env.rs

```diff
@@ -423,13 +419,14 @@ impl Env {
         contract_id
     }
 
-    /// Register a contract in a WASM file with the [Env] for testing.
+    /// Install the contract WASM code to the [Env] for testing.
     ///
-    /// Passing a contract ID for the first arguments registers the contract
-    /// with that contract ID. Providing `None` causes a random ID to be
-    /// assigned to the contract.
+    /// Returns the identifier of the installed code that can be then used for
+    /// the contract deployment.
     ///
-    /// Returns the contract ID of the registered contract.
+    /// This is mostly useful for contract factory testing, otherwise use
+    /// `register_contract_wasm` function that installs and deploys the contract
+    /// in a single call.
```

Comment:
Avoid "This" in docs. It can be used in every paragraph/sentence and makes for more succinct explanations and docs.
```suggestion
    /// Useful for contract factory testing, otherwise use
    /// `register_contract_wasm` function that installs and deploys the contract
    /// in a single call.
```

Reply:
Updated.